components/brave_vpn/common/brave_vpn_utils.cc

```diff
@@ -44,6 +44,9 @@ void RegisterVPNLocalStatePrefs(PrefRegistrySimple* registry) {
 #if BUILDFLAG(ENABLE_BRAVE_VPN_WIREGUARD)
   registry->RegisterBooleanPref(prefs::kBraveVPNWireguardEnabled, false);
 #endif
+#if BUILDFLAG(IS_MAC)
+  registry->RegisterBooleanPref(prefs::kBraveVPNOnDemandEnabled, false);
```

Comment:
Out of curiosity, unlike other prefs, this seems to be a choice of users. Should this be ProfilePref?

Reply:
It should be local state because os vpn entry is per application(not per profile).
vpn entry is used on all profiles.

Reply:
I see. Thanks!